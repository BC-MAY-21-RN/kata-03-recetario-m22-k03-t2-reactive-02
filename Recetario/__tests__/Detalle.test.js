import 'react-native';
import React from 'react';
import Detalle from '../src/screens/detalle';

const product = {
  id: 1,
  title: 'Lemon Pie',
  img: 'https://t2.rg.ltmcdn.com/es/posts/2/7/7/lemon_pie_saludable_75772_600.jpg',
  ingredients: [
    {name: 'egg white', amount: '1'},
    {name: 'wholemeal flour', amount: '70 grams'},
    {name: 'oatmeal', amount: '30 grams'},
    {name: 'sugar', amount: '1 tablespoon'},
    {name: 'yolk', amount: '1'},
    {
      name: 'skim milk or unsweetened vegetable milk',
      amount: '200 milliliters',
    },
    {name: 'lemon juice and zest', amount: '1/2 juice'},
  ],
  like: false,
  servings: '6',
};

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(<Detalle route={{params: product}} />).toJSON();
  expect(tree).toMatchSnapshot();
});