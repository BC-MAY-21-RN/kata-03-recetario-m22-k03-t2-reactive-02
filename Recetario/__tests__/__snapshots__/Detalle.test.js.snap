// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#282929",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "height": "50%",
        "marginBottom": 15,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://t2.rg.ltmcdn.com/es/posts/2/7/7/lemon_pie_saludable_75772_600.jpg",
        }
      }
      style={
        Object {
          "height": "100%",
          "opacity": 0.5,
          "position": "absolute",
          "width": "100%",
        }
      }
    />
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "margin": 10,
          "marginTop": 15,
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#f4f5f4",
                "fontSize": 50,
              },
              undefined,
              Object {
                "fontFamily": "EvilIcons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#f4f5f4",
                  "fontSize": 50,
                },
                undefined,
                Object {
                  "fontFamily": "EvilIcons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#f4f5f4",
                  "fontSize": 50,
                },
                undefined,
                Object {
                  "fontFamily": "EvilIcons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
    <Text
      style={
        Object {
          "color": "#dfdfde",
          "fontSize": 20,
          "fontWeight": "bold",
          "marginLeft": 20,
          "marginTop": "50%",
        }
      }
    >
      Lemon Pie
    </Text>
    <Text
      style={
        Object {
          "color": "#dfdfde",
          "fontSize": 15,
          "fontWeight": "bold",
          "marginLeft": 20,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "marginLeft": 20,
        "marginRight": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#dfdfde",
          "fontSize": 15,
          "fontWeight": "bold",
        }
      }
    >
      Ingredients
    </Text>
    <Text
      style={
        Object {
          "color": "#dfdfde",
          "fontSize": 15,
          "fontWeight": "bold",
        }
      }
    >
      for 
      6
       servings
    </Text>
    <RCTScrollView
      data={
        Array [
          Object {
            "amount": "1",
            "name": "egg white",
          },
          Object {
            "amount": "70 grams",
            "name": "wholemeal flour",
          },
          Object {
            "amount": "30 grams",
            "name": "oatmeal",
          },
          Object {
            "amount": "1 tablespoon",
            "name": "sugar",
          },
          Object {
            "amount": "1",
            "name": "yolk",
          },
          Object {
            "amount": "200 milliliters",
            "name": "skim milk or unsweetened vegetable milk",
          },
          Object {
            "amount": "1/2 juice",
            "name": "lemon juice and zest",
          },
        ]
      }
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      style={
        Object {
          "flexGrow": 0,
          "height": 250,
          "marginTop": 10,
        }
      }
      viewabilityConfigCallbackPairs={Array []}
    >
      <View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "borderBottomColor": "grey",
                "borderBottomWidth": 0.6,
                "borderStyle": "solid",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "paddingBottom": 15,
                "paddingTop": 15,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#dfdfde",
                  "fontWeight": "bold",
                  "marginLeft": 5,
                }
              }
            >
              egg white
            </Text>
            <Text
              style={
                Object {
                  "color": "#dfdfde",
                  "fontWeight": "bold",
                  "marginLeft": 5,
                }
              }
            >
              1
            </Text>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "borderBottomColor": "grey",
                "borderBottomWidth": 0.6,
                "borderStyle": "solid",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "paddingBottom": 15,
                "paddingTop": 15,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#dfdfde",
                  "fontWeight": "bold",
                  "marginLeft": 5,
                }
              }
            >
              wholemeal flour
            </Text>
            <Text
              style={
                Object {
                  "color": "#dfdfde",
                  "fontWeight": "bold",
                  "marginLeft": 5,
                }
              }
            >
              70 grams
            </Text>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "borderBottomColor": "grey",
                "borderBottomWidth": 0.6,
                "borderStyle": "solid",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "paddingBottom": 15,
                "paddingTop": 15,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#dfdfde",
                  "fontWeight": "bold",
                  "marginLeft": 5,
                }
              }
            >
              oatmeal
            </Text>
            <Text
              style={
                Object {
                  "color": "#dfdfde",
                  "fontWeight": "bold",
                  "marginLeft": 5,
                }
              }
            >
              30 grams
            </Text>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "borderBottomColor": "grey",
                "borderBottomWidth": 0.6,
                "borderStyle": "solid",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "paddingBottom": 15,
                "paddingTop": 15,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#dfdfde",
                  "fontWeight": "bold",
                  "marginLeft": 5,
                }
              }
            >
              sugar
            </Text>
            <Text
              style={
                Object {
                  "color": "#dfdfde",
                  "fontWeight": "bold",
                  "marginLeft": 5,
                }
              }
            >
              1 tablespoon
            </Text>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "borderBottomColor": "grey",
                "borderBottomWidth": 0.6,
                "borderStyle": "solid",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "paddingBottom": 15,
                "paddingTop": 15,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#dfdfde",
                  "fontWeight": "bold",
                  "marginLeft": 5,
                }
              }
            >
              yolk
            </Text>
            <Text
              style={
                Object {
                  "color": "#dfdfde",
                  "fontWeight": "bold",
                  "marginLeft": 5,
                }
              }
            >
              1
            </Text>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "borderBottomColor": "grey",
                "borderBottomWidth": 0.6,
                "borderStyle": "solid",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "paddingBottom": 15,
                "paddingTop": 15,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#dfdfde",
                  "fontWeight": "bold",
                  "marginLeft": 5,
                }
              }
            >
              skim milk or unsweetened vegetable milk
            </Text>
            <Text
              style={
                Object {
                  "color": "#dfdfde",
                  "fontWeight": "bold",
                  "marginLeft": 5,
                }
              }
            >
              200 milliliters
            </Text>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "borderBottomColor": "grey",
                "borderBottomWidth": 0.6,
                "borderStyle": "solid",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "paddingBottom": 15,
                "paddingTop": 15,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#dfdfde",
                  "fontWeight": "bold",
                  "marginLeft": 5,
                }
              }
            >
              lemon juice and zest
            </Text>
            <Text
              style={
                Object {
                  "color": "#dfdfde",
                  "fontWeight": "bold",
                  "marginLeft": 5,
                }
              }
            >
              1/2 juice
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
